model/flow/identity_test.go

```diff
@@ -70,6 +72,74 @@ func TestIdentityEncodingMsgpack(t *testing.T) {
 }
 */
 
+func TestIdentityList_Exists(t *testing.T) {
+	t.Run("should find a given elemen", func(t *testing.T) {
+		il1 := unittest.IdentityListFixture(10)
+		il2 := unittest.IdentityListFixture(1)
+
+		// sort the first list
+		sort.Slice(il1, func(p, q int) bool {
+			num1 := il1[p].NodeID[:]
+			num2 := il1[q].NodeID[:]
+			lenID := len(num1)
+
+			// assume the length is a multiple of 4, for performance.  it's 32 bytes
+			for i := 0; i < lenID; i += 8 {
+				chunk1 := binary.BigEndian.Uint64(num1[i:])
+				chunk2 := binary.BigEndian.Uint64(num2[i:])
+
+				if chunk1 < chunk2 {
+					return true
+				} else if chunk1 == chunk2 {
+					continue
+				} else {
+					return false
+				}
+			}
+			// equality
+			return true
+		})
+		element := il1[8]
```

Comment:
instead of picking one element, it's better to ensure that every element exists.

I would suggest to make the test cases bigger:
create a identity list with 500 identities (which is close to mainnet), go through each identity and verify they exist

Reply:
will do